docs/how-to-guides/admin/gitlab.rst

```diff
@@ -0,0 +1,240 @@
+  https://<your-renku-dns>/api/auth/gitlab/token
+
+Scopes:
+
+.. code-block:: console
+
+    api (Access the authenticated user's API)
+    read_user (Read the authenticated user's personal information)
+    read_repository (Allows read-access to the repository)
+    read_registry (Grants permission to read container registry images)
+    openid (Authenticate using OpenID Connect)
+
+Copy the secret and client ID for use in the step below.
+
+Configuring an External GitLab as Identity Provider for Renku
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If the Renku deployment is relying on an external GitLab instance, 
+this GitLab instance could be configured as an identity provider for Renku's Keycloak. 
+This way the Renku login screen will show the option to "Login via GitLab" 
+and existing GitLab users can use Renku without creating a separate Renku account.
+
+#. Login to the admin console of Renku's Keycloak, usually found at \https://<renku-domain>/auth, using the username and password stored in the Kubernetes secret called  ``keycloak-password-secret``. Ensure you ``base64`` decode these values before using them to login. If you use an external Keycloak with Renku, login to the external Keycloak instead.
+#. Add an ``Identity Provider`` of type ``OpenID Connect v1.0``.
+#. Set ``Alias`` to <renku-domain>, ``Authorization URL`` to \https://<gitlab-domain>/oauth/authorize, ``Token URL`` to \https://<gitlab-domain>/oauth/token and ``Client ID`` and ``Client Secret`` to the respective values copied from the Renku GitLab application details.
+#. Click ``Save``. The new Identity Provider should appear and any user from the stand-alone GitLab instance should be able to login.
+
+GitLab deployed as part of Renku
+--------------------------------
+
+We do *not* recommend deploying the Renku-bundled GitLab as part of a production Renku deployment, 
+and instead suggest deploying GitLab using the `official GitLab cloud-native Kubernetes chart 
+<https://docs.gitlab.com/charts/>`_. Deploying GitLab as part of Renku may be deprecated in the future.
+
+If your Renku deployment includes GitLab you need to follow some additional steps to configure an admin user on GitLab.
+
+To grant a GitLab user the GitLab admin role without having access to the GitLab Web UI, the following steps can be taken in the GitLab container console.
+
+#. Run ``gitlab-rails console -e production`` (this might take a while).
+#. Find the user you would like to grant the admin role, for example by running ``user = User.find_by(email: 'renku@renkulab.io')`` or ``user = User.find_by(username: 'renku')``.
+#. Grant the user the administrator role by running ``user.admin = true``.
+#. Save the user's profile by running ``user.save!``.
+#. Leave the console by running ``exit``.
+
+Migrate from Renku-bundled Omnibus GitLab to cloud-native Gitlab Helm chart
+---------------------------------------------------------------------------
+
```

Comment:
```suggestion

Migrating from the Renku-bundled Gitlab to the cloud the cloud-native Helm chart should be done only after the existing Gitlab deployment is fully and properly backed up. Doing the migration without having a fresh backup of the existing Gitlab has a very high risk of irrecoverable loss of user data and code.

```

Reply:
I'm not sure we need to add this, as the backup is part of the migration and is covered on lines 165-187